From 17ba7b80b78795e9ee6ff59f6f85178b5d155d3f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?= <avarab@gmail.com>
Date: Mon, 13 Sep 2010 12:42:29 +0000
Subject: [PATCH 109/161] Makefile: only add gettext tests on XGETTEXT_INCLUDE_TESTS=YesPlease
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Change the "pot" target so that the xgettext(1) invocation no longer
include test files from t/t0200.

These should only be included in the Icelandic is.po file since Git's
own test suite uses them for regression tests, but nobody else should
translate them. Previously I indicated this with a TRANSLATORS comment
for all the strings:

    #. TRANSLATORS: This is a test. You don't need to translate it.
    #, c-format
    msgid "TEST: A C test string %s"
    msgstr ""

But if translators follow those instructions they'll never have 100%
translation coverage. It's much better to add a hidden option so that
is.po now has to use a special git.pot file for msgmerge(1):

    make pot XGETTEXT_INCLUDE_TESTS=YesPlease

But everything else can continue to use the documented:

    make pot

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 Makefile |   12 +++++++++---
 1 files changed, 9 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index 4064df1..e1650ba 100644
--- a/Makefile
+++ b/Makefile
@@ -2073,9 +2073,15 @@ XGETTEXT_OPTIONS_C = $(XGETTEXT_OPTIONS) --keyword=_ --keyword=N_ --language=C
 XGETTEXT_OPTIONS_SH = $(XGETTEXT_OPTIONS) --language=Shell
 XGETTEXT_OPTIONS_PERL = $(XGETTEXT_OPTIONS) --keyword=__ --language=Perl
 
-LOCALIZED_C = $(C_OBJ:o=c) t/t0200/test.c
-LOCALIZED_SH = $(SCRIPT_SH) t/t0200/test.sh
-LOCALIZED_PERL = $(SCRIPT_PERL) t/t0200/test.perl
+LOCALIZED_C = $(C_OBJ:o=c)
+LOCALIZED_SH = $(SCRIPT_SH)
+LOCALIZED_PERL = $(SCRIPT_PERL)
+
+ifdef XGETTEXT_INCLUDE_TESTS
+LOCALIZED_C += t/t0200/test.c
+LOCALIZED_SH += t/t0200/test.sh
+LOCALIZED_PERL += t/t0200/test.perl
+endif
 
 pot:
 	$(XGETTEXT) $(XGETTEXT_OPTIONS_C) $(LOCALIZED_C)
-- 
1.7.3.1.50.g1e633

